src/ReverseRegex/Random/SimpleRandom.php

```diff
@@ -94,27 +94,27 @@ public function seed($seed = null)
       *  Generate a random numer
       *
       *  @param integer $max
-      *  @param integer $max 2,796,203 largest possible max
+      *  @param integer $max will use PHP_INT_MAX for largest possible max
       */
     public function generate($min = 0, $max = null)
     {  
         if($max === null) {
-            $max = 2796203;
+            $max = PHP_INT_MAX;
         }
         
-        if($max > 2796203) {
+        if($max > PHP_INT_MAX) {
             throw new ReverseRegexException('Max param has exceeded the maxium 2796203');
         }
         
         if ($this->seed == 0) {
             $this->seed(mt_rand());
         }
         
-        $this->seed = ($this->seed * 125) % 2796203;  
+        $this->seed = ($this->seed * 125) % PHP_INT_MAX;  
```

Comment:
This causes problems. If I use your code, the results when calling the parser become unpredictable. E.g. the example regex of `[a-z]{10}` doesn't produce 10 characters. It'll be some random number of characters less than 10.